matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/pushers/DefaultPushersService.kt

```diff
@@ -66,27 +66,39 @@ internal class DefaultPushersService @Inject constructor(
                                deviceDisplayName: String,
                                url: String,
                                append: Boolean,
-                               withEventIdOnly: Boolean)
-            : UUID {
-        // Do some parameter checks. It's ok to throw Exception, to inform developer of the problem
-        if (pushkey.length > 512) throw InvalidParameterException("pushkey should not exceed 512 chars")
-        if (appId.length > 64) throw InvalidParameterException("appId should not exceed 64 chars")
-        if ("/_matrix/push/v1/notify" !in url) throw InvalidParameterException("url should contain '/_matrix/push/v1/notify'")
-
-        val pusher = JsonPusher(
-                pushKey = pushkey,
-                kind = "http",
-                appId = appId,
-                appDisplayName = appDisplayName,
-                deviceDisplayName = deviceDisplayName,
-                profileTag = profileTag,
-                lang = lang,
-                data = JsonPusherData(url, EVENT_ID_ONLY.takeIf { withEventIdOnly }),
-                append = append)
+                               withEventIdOnly: Boolean
+    ) = addPusher(
+            JsonPusher(
+                    pushKey = pushkey,
+                    kind = "http",
+                    appId = appId,
+                    profileTag = profileTag,
+                    lang = lang,
+                    appDisplayName = appDisplayName,
+                    deviceDisplayName = deviceDisplayName,
+                    data = JsonPusherData(url, EVENT_ID_ONLY.takeIf { withEventIdOnly }),
+                    append = append
+            )
+    )
 
-        val params = AddHttpPusherWorker.Params(sessionId, pusher)
+    override fun addEmailPusher(email: String, lang: String, emailBranding: String, appDisplayName: String, deviceDisplayName: String) = addPusher(
```

Comment:
One param per line is better when there are a lot of parameters like here

Reply:
Also why there is no param `append` here?

Reply:
`append` was removed as an option because it sounds like it has the side effect of causing other accounts which use the same email to lose email notifications https://github.com/matrix-org/matrix-react-sdk/pull/2727/files#diff-ec232520bf51337e5e6939b885d21f428ad6da3306c8e17a3ff660b2b341179dR165

Reply:
It's sometimes difficult to decide between what we want the SDK to expose (as parameters for instance) and what we want the SDK to handle, so keep hidden from the client app. On one side the SDK should let the app do whatever it wants to, and on the other side we want the SDK to provide a high level service... So maybe add the parameter, with default value to `true` and with the same comment from https://github.com/matrix-org/matrix-react-sdk/pull/2727/files#diff-ec232520bf51337e5e6939b885d21f428ad6da3306c8e17a3ff660b2b341179dR165 in the doc of the fun `addEmailPusher`.
Thanks!

Reply:
makes sense, will do :+1: